src/table-doc.js

```diff
@@ -0,0 +1,57 @@
+import { escapeHTML } from './strings';
+
+export function tableDoc(code) {
+    try {
+        let json = JSON.parse(code);
+        // support language specific overrides
+        // if (json.languages && json.langauges[language]) {
+        //     Object.assign(json, json.languages[language]);
+        // }
+        
+        const td = [];
+
+        td.push(`<table>`);
+        td.push(`<tr><th>Name</th><th>Type</th></tr>`);
+        
+        if(json.columns){
+            td.push(tableHeaders(json.columns));
+        }
+        // if(json.example){
+        //     td.push(tableData(json.data));
+        // }
+
+        td.push(`</table>`);
+        console.log(td.join('\n'));
+        return td.join('\n');
+        
+    }
+    catch (ex) {
+        console.log(ex);
+        return '`failed to render %jsonblock: `' + ex.message + '`';
+    }
+}
+
+function tableData(json){
+    let data = [];
+    json.forEach(function(elem, i, arr) {
+        data[i] = elem.map((a) => `<td>${a}</td>`);
+        data[i].unshift(`<tr>`);
+        data[i].push(`</tr>`);
+    });
+    return data.map((a) => a.join('')).join('\n');
```

Comment:
Instead of doing `data.map((a) => a.join(''))`, you can do that inside `forEach` so that it's done in single pass. Also, doing that will eliminate the need to do `unshift`/`push`.
```javascript
data[i] = '<tr>' + elem.map(a => `<td>${a}</td>`).join('') + '</tr>';
```

This works, but it'll be better to use `map` instead of `forEach` in this case because you're purely transforming `json`.
```javascript
json.map(elem => '<tr>' + elem.map(a => `<td>${a}</td>`).join('') + '</tr>').join('\n');
```